libraries/Hash/src/Hash.cpp

```diff
@@ -78,20 +75,21 @@ void sha1(const char * data, uint32_t size, uint8_t hash[20]) {
     sha1((uint8_t *) data, size, hash);
 }
 
-void sha1(String data, uint8_t hash[20]) {
+void sha1(const String& data, uint8_t hash[20]) {
     sha1(data.c_str(), data.length(), hash);
 }
 
 String sha1(uint8_t* data, uint32_t size) {
     uint8_t hash[20];
-    String hashStr = "";
+    String hashStr((const char*)nullptr);
+    hashStr.reserve(20 * 2 + 1);
 
     sha1(&data[0], size, &hash[0]);
 
     for(uint16_t i = 0; i < 20; i++) {
         String hex = String(hash[i], HEX);
-        if(hex.length() < 2) {
-            hex = "0" + hex;
+        if(hash[i] < 0x10) {

```

Comment:
I know you just did a quick change, but the original was written in a weird way.  Can we make it a little cleaner?
````
for(uint16_t i = 0; i < 20; i++) {
    char hex[3];
    snprintf(hex, sizeof(hex), "%02x", hash[i]);
    hashStr += hex;
}
````

Reply:
I wonder why we wouldn't do a single sprintf call then using 20 parameters instead of 20 in a loop?

Reply:
Code size will be larger, probably.  And, less typing or chance of typo. :)

Reply:
WFM